subprojects/instant-execution/src/main/kotlin/org/gradle/instantexecution/fingerprint/InstantExecutionCacheFingerprintController.kt

```diff
@@ -126,14 +128,15 @@ class InstantExecutionCacheFingerprintController internal constructor(
 
     private
     fun addListener(listener: InstantExecutionCacheFingerprintWriter) {
-        listenerManager.service.addListener(listener)
+        listenerManager.addListener(listener)
+        buildTreeListenerManager.service.addListener(listener)
         taskInputsListeners.addListener(listener)
     }
 
     private
     fun removeListener(listener: InstantExecutionCacheFingerprintWriter) {
         taskInputsListeners.removeListener(listener)
-        listenerManager.service.removeListener(listener)
```

Comment:
❓ Hey, @adammurdoch, shouldn't this be `listenerManager.removeListener(listener)`?